se/se_epub_build.py

```diff
@@ -82,8 +83,8 @@ def build(self, metadata_xhtml: str, metadata_tree: se.easy_xml.EasyXmlTree, run
 	if verbose:
 		print(f"Building {self.path} ...")
 
-	with tempfile.TemporaryDirectory() as work_directory:
-		work_directory = Path(work_directory)
+	with tempfile.TemporaryDirectory() as temp_directory:
```

Comment:
Here we change to a different name and then reuse that name in line 510. But refs to `work_directory` are still around (line 924) and if we reuse the name `temp_directory` in line 510 won't that destroy the previous reference? Why not leave these names as they are?

Reply:
The "golden rule" to keep in mind with mypy is that if you use a local variable for one type, you can't reuse it later for another type or mypy will complain. The value assigned to `temp_directory` is type `str`. The value assigned to `work_directory` is type `Path`. That's why we introduced a second name -- so that `work_directory` is not reused with a different type.

As for reusing the `temp_directory` name later in the function, that is ok with mypy because the type is the same in the new usage (i.e. type `str`). And it also ok from a correctness point of view because old value is effectively copied when it is passed to the `Path` constructor so the value of `temp_directory` can change without affecting the value of `work_directory`.